src/axom/core/tests/core_device_hash.hpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+// Axom includes
+#include "axom/config.hpp"
+#include "axom/core/Macros.hpp"
+#include "axom/core/DeviceHash.hpp"
+
+// gtest includes
+#include "gtest/gtest.h"
+
+TEST(core_device_hash, hash_int)
```

Comment:
DeviceHash is being tested here, but not on various devices. Is that demonstrated elsewhere such as in FlatMap tests?

Reply:
I think that was originally the idea. We could add additional testing to this PR as well.